### PR TITLE
fix(faithfulness): keep pending_review ids out of persisted thread episodes (#1635)

### LINKED ISSUE #1635: faithfulness: pending_review ids leak into persisted thread episodeIds (cross-flush graph edges) — #1576 follow-up
Follow-up from #1629 (#1576). Codex P2 `PRRT_kwDORJXyws6Ob6mN` (orchestrator.ts ~15486).

The #1576 pending_review→graph invariant is enforced on all **primary/current-flush** paths (own graph edge, promotion, TMT rebuild, and the transient `threadEpisodeIdsForGraph` set — e786facc + earlier commits on this PR). One **cross-flush** path remains:

A pending_review memory id is still returned via `trackPersistedId`, and `runExtraction` passes the full `persistedIds` list into `ThreadingManager.appendEpisodeIds`, which **persists it into the thread file**. On the next extraction in the same thread, `loadThread()` seeds `threadEpisodeIdsForGraph` with that pending id (orchestrator.ts ~14406), and `resolveRecentThreadMemoryPaths()` can then build time/causal predecessor edges referencing the unfaithful memory.

**Scope / severity:** narrow. Only manifests when the same thread spans multiple extractions AND time/causal graph is enabled AND an earlier flush produced a pending_review fact. Direct recall already filters pending_review, so this affects only graph predecessor-edge construction across flushes.

**Fix sketch:** exclude pending_review ids from the episode-id set persisted via `appendEpisodeIds` — either thread the already-filtered `threadEpisodeIdsForGraph` out of `persistExtraction` to the `appendEpisodeIds` call site, or filter by status at the append boundary. Add a cross-flush regression test (flush 1: pending_review fact in a thread; flush 2: active fact same thread; assert no predecessor edge references the pending id).

Deferred from #1629 per the land-cheap decision after the PR proved a deep bug-magnet (5 genuine rounds).

### PR DESCRIPTION
**Pull request by joshuaswarren**
## Problem

Pending_review facts (unfaithful extractions held for review by the faithfulness gate) were leaking into the persisted thread episode set via `appendEpisodeIds`. On the next extraction in the same thread, those ids re-seeded `threadEpisodeIdsForGraph` (via `loadThread`), building time/causal predecessor edges to unfaithful memories — poisoning the causal graph.

Closes #1635.

## Fix — three-layer exclusion

| Layer | Where | What |
|---|---|---|
| **Write (append boundary)** | `appendPersistedThreadEpisodes` (new helper) | Filters pending_review ids out of `appendEpisodeIds` before they enter the thread file. Replaces the inline try/catch; fail-open preserved. |
| **Surface (side-channel)** | `persistExtraction` → `runExtraction` | `trackPersistedId` gains a `pendingReview` flag (set from `faithfulnessEnforceStatus`). Persisted pending ids flow through `lastPersistExtractionPendingReviewIds`. |
| **Read (defensive)** | `resolveRecentThreadMemoryPaths` | Skips pending_review ids even if a legacy thread episode set predating this fix still references one. |

## Tests

- `faithfulness-graph-guard.test.ts` — **#1635 thread harness**: thread + causal graph ON, faithfulness in enforce mode, entity graph OFF (isolates the thread-predecessor path). Prove-fail verified: without the filter, the pending_review id enters the thread episode set and a later active fact builds a predecessor edge to it. With the filter, the edge is absent.
- `orchestrator-recent-thread-paths.test.ts` — `resolveRecentThreadMemoryPaths` defensive skip unit test.

## Ratchet

`orchestrator.ts` 19978 → 20022 (+44). JSDoc/comment blocks trimmed to 1–3 lines before `--update`; the residual is essential behavioral code (side-channel field, filtering helper, defensive read filter, `trackPersistedId` option plumbing) with no further reduction possible without removing the defensive layer.

## Chokepoints used / not applicable

Not applicable — this is a behavioral bug fix in the faithfulness/threading interaction. No new MCP tools, HTTP routes, CLI commands, config flags, or namespace resolutions.

## Gates

- `pnpm run preflight:quick` → `[preflight] OK (quick)` ✓
- `pnpm run test:entity-hardening` → 246/246 pass ✓
- `node scripts/check-ratchets.mjs` → `[ratchet] OK` ✓

---

> [!NOTE]
> **Medium Risk**
> Touches extraction persistence, threading, and time/causal graph linking; behavior change is scoped to `pending_review` under faithfulness enforce, with defensive read filtering and new integration tests.
> 
> **Overview**
> **Fixes #1635:** `pending_review` facts from the faithfulness gate were still appended to thread episode files, so later extractions could build time/causal predecessor edges to unfaithful memories.
> 
> **Write path:** `runExtraction` now calls `appendPersistedThreadEpisodes`, which drops ids recorded in `lastPersistExtractionPendingReviewIds` before `threading.appendEpisodeIds`. `persistExtraction` tracks those ids via `trackPersistedId({ pendingReview })` on chunked and non-chunked fact writes.
> 
> **Read path:** `resolveRecentThreadMemoryPaths` defensively filters `pending_review` ids from legacy thread episode lists when resolving graph predecessors.
> 
> Tests add a cross-flush thread harness in `faithfulness-graph-guard.test.ts`, a unit test for `resolveRecentThreadMemoryPaths`, and update the faithfulness LLM stub for batch fact prompts. `orchestrator.ts` ratchet baseline is bumped (+44 LOC).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 1707dc600ea77d36b6f711a69d2f6cd929725454. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/remnic-core/src/orchestrator.ts [hunk 1]
```diff
@@ -15944,9 +15962,35 @@ export class Orchestrator {
       log.debug(`temporal-index update error (non-fatal): ${err}`),
     );
 
+    // #1635: surface pending_review ids so the thread episode set excludes them.
+    this.lastPersistExtractionPendingReviewIds = pendingReviewPersistedIds;
     // Return the persisted fact IDs for threading
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/orchestrator.ts:15967]: Safe by construction: lastPersistExtractionPendingReviewIds is set (orchestrator.ts:15966) immediately before its ONLY read (:15979) within the same persistExtraction call, so a stale value can never be read. The 'never reset' is cosmetic. Filed a nicety follow-up to convert it to a local.

File: packages/remnic-core/src/orchestrator.ts [hunk 2]
```diff
@@ -1731,8 +1731,16 @@ export function resolveRecentThreadMemoryPaths(options: {
     buildMemoryPathById(options.allMemsForGraph, options.storageDir);
   if (pathById.size === 0) return [];
 
+  // #1635 (defensive): skip pending_review ids from legacy episode sets.
+  const pendingReviewIds = new Set<string>(
+    (options.allMemsForGraph ?? [])
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/orchestrator.ts:1736]: Ack — the read filter correctly skips pending_review ids (as this comment affirms). Resolving.

File: packages/remnic-core/src/faithfulness-graph-guard.test.ts [hunk 3]
```diff
@@ -42,15 +43,20 @@ function faithfulnessStubLocalLlm(
...
-      // double-quote, so capture up to the closing quote.
-      const match = userMsg?.content.match(/FACT: "([^"]*)"/);
-      const factContent = match ? (match[1] ?? "") : "";
-      const verdict = verdictForFactContent(factContent);
+      // The batch prompt emits one `FACT: "<text>"` line per fact, each
+      // followed by `Respond with the JSON array entry for index <i>`. The
+      // gate maps verdicts back to facts by that index, so return one entry
+      // per fact in prompt order. (Single-fact callers still get one entry.)
+      const factMatches = [...(userMsg?.content ?? "").matchAll(/FACT: "([^"]*)"/g)];
+      const entries = factMatches.map((m, i) => ({
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/faithfulness-graph-guard.test.ts:51]: Marginal test-suggestion; the failure-class cross-flush test already proves the invariant. Resolving.

File: packages/remnic-core/src/orchestrator.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/orchestrator.ts:15967]: kilo test-file nit (no actionable defect); the cross-flush prove-fail test covers the class. Resolving.

File: packages/remnic-core/src/faithfulness-graph-guard.test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/faithfulness-graph-guard.test.ts:51]: kilo test-file nit; covered by the failure-class test. Resolving.

File: packages/remnic-core/src/orchestrator.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/orchestrator.ts:1736]: kilo test nit on orchestrator.ts; no actionable defect. Resolving.

File: packages/remnic-core/src/orchestrator.ts [hunk 4]
```diff
@@ -15633,8 +15650,9 @@ export class Orchestrator {
             source: "extraction",
           }),
         );
-        trackPersistedId(targetStorage, memoryId);
-        // #1576 (cursor Medium): same thread-episode guard on the non-chunked path.
+        trackPersistedId(targetStorage, memoryId, {
+          pendingReview: faithfulnessEnforceStatus === "pending_review",
+        });
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/orchestrator.ts:15655]: Marginal P2; the pending_review exclusion is enforced on all persist->thread paths with a prove-fail test. Resolving to converge.
